From 3f5df45da89148581e096714ca261e32cfc34550 Mon Sep 17 00:00:00 2001
From: Song Bing <b06498@freescale.com>
Date: Tue, 9 Oct 2012 14:56:58 +0800
Subject: [PATCH 01/68] ENGR00133115 Fix MediaPlayerTest CTS fai.

android.media.cts.MediaPlayerTest#testRecordedVideoPlayback0
android.media.cts.MediaPlayerTest#testRecordedVideoPlayback270
H263 encoder component will use timestamp queue after delete it. which will cause random timestamp value.
MediaPlayerTest is fixed by delete timestamp queue after use it in 263 encode component.

Signed-off-by: Song Bing b06498@freescale.com
---
 media/libstagefright/codecs/m4v_h263/enc/SoftMPEG4Encoder.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/media/libstagefright/codecs/avc/enc/SoftAVCEncoder.cpp b/media/libstagefright/codecs/avc/enc/SoftAVCEncoder.cpp
old mode 100644
new mode 100755
diff --git a/media/libstagefright/codecs/m4v_h263/enc/SoftMPEG4Encoder.cpp b/media/libstagefright/codecs/m4v_h263/enc/SoftMPEG4Encoder.cpp
old mode 100644
new mode 100755
index e02af90..955022e
--- a/media/libstagefright/codecs/m4v_h263/enc/SoftMPEG4Encoder.cpp
+++ b/media/libstagefright/codecs/m4v_h263/enc/SoftMPEG4Encoder.cpp
@@ -754,6 +754,7 @@ void SoftMPEG4Encoder::onQueueFilled(OMX_U32 portIndex) {
         mInputBufferInfoVec.erase(mInputBufferInfoVec.begin());
         outInfo->mOwnedByUs = false;
         notifyFillBufferDone(outHeader);
+        mInputBufferInfoVec.erase(mInputBufferInfoVec.begin());
     }
 }
 
-- 
1.8.0

